src/syrupy/assertion.py

```diff
@@ -91,12 +78,11 @@ def __eq__(self, other) -> bool:
     def _assert(self, data) -> bool:
         self._session.register_assertion(self)
         try:
-            serialized_data = self.serializer.encode(data)
```

Comment:
I think we mistakenly reverted the checking without deserializing in this PR 🤦‍♂